lib/ecto/repo/model.ex

```diff
@@ -119,9 +119,18 @@ defmodule Ecto.Repo.Model do
   defp struct_from_changeset!(%{model: struct}),
     do: struct
 
-  defp load_into_changeset(%{changes: changes} = changeset, model, return, values) do
-    update_in changeset.model,
-              &model.__schema__(:load, struct(&1, changes), return, values)
+  defp load_into_changeset(%{changes: changes} = changeset, model, values) do
+
+    update_in changeset.model, &do_load(struct(&1, changes), model, values)
+  end
+
+  defp do_load(struct, model, kv) do
+    loaded = Enum.reduce(kv, {struct}, fn
+      {k,v}, {acc} ->
+        value = Ecto.Type.load!(model.__schema__(:field, k), v)
+        {Map.put(acc, k, value)}
+    end) |> elem(0) |> Map.put(:__state__, :loaded)
+    Ecto.Model.Callbacks.__apply__(struct.__struct__, :after_load, loaded)
```

Comment:
Not sure this callback call here is needed, but there is a test expecting this call.